rollout/trafficrouting.go

```diff
@@ -148,6 +148,8 @@ func (c *rolloutContext) reconcileTrafficRouting() error {
 				// weight of the traffic routing service should be at the value of the
 				// last setWeight step, which is set by GetCurrentSetWeight.
 				desiredWeight = replicasetutil.GetCurrentSetWeight(c.rollout)
+			} else if *index == int32(len(c.rollout.Spec.Strategy.Canary.Steps)) {
```

Comment:
I think this is contradicting to the purpose of 

https://github.com/argoproj/argo-rollouts/blob/200b045320b7202a07f727c558f2c0f2f703f8af/rollout/trafficrouting.go#L143-L151

Instead, we should investigate why replicasetutil.GetCurrentSetWeight(c.rollout) doesn't return 100.

https://github.com/argoproj/argo-rollouts/blob/200b045320b7202a07f727c558f2c0f2f703f8af/utils/replicaset/canary.go#L442-L449

Reply:
Setting the last step weight to 100 will achieve the desired weight, but changing the stable service selector after the VS route occurs will invalidate the VS, and traffic will be sent to old and new pods. Old pods receive traffic during terminate, resulting ina large number of 503 HTTP codes. The correct order would be to update stable service Selector first and wait for istio xDS to update. Then switch 100% of the traffic to stable service.
If destinationrule subsets are used, two subsets using the same labels will also invalidate the VS rule.

Reply:
Yes, but I can't find a reason to prevent desireweight from being set to 100